sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -320,10 +320,15 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
    * Create a [[DescribeTableCommand]] logical plan.
    */
   override def visitDescribeTable(ctx: DescribeTableContext): LogicalPlan = withOrigin(ctx) {
-    // Describe column are not supported yet. Return null and let the parser decide
-    // what to do with this (create an exception or pass it on to a different system).
     if (ctx.describeColName != null) {
-      null
+      if (ctx.partitionSpec != null) {
+        throw new ParseException("DESC TABLE COLUMN for a specific partition is not supported", ctx)
+      } else {
+        DescribeColumnCommand(
+          visitTableIdentifier(ctx.tableIdentifier),
+          ctx.describeColName.getText,
+          ctx.FORMATTED != null)
```

Comment:
I just tested in hive 2.1, it only show detailed column info with `formatted`, shall we keep the same as hive?
```
hive> desc test key;
OK
key                 	int                 	from deserializer   
Time taken: 0.063 seconds, Fetched: 1 row(s)
hive> desc extended test key;
OK
key                 	int                 	from deserializer   
Time taken: 0.053 seconds, Fetched: 1 row(s)
hive> desc formatted test key;
OK
# col_name            	data_type           	min                 	max                 	num_nulls           	distinct_count      	avg_col_len         	max_col_len         	num_trues           	num_falses          	comment             
	 	 	 	 	 	 	 	 	 	 
key                 	int                 	1                   	2                   	0                   	2                   	                    	                    	                    	                    	from deserializer   
Time taken: 0.067 seconds, Fetched: 3 row(s)
```